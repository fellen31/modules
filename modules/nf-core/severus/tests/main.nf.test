nextflow_process {

    name "Test Process SEVERUS"
    script "../main.nf"
    process "SEVERUS"

    tag "modules"
    tag "modules_nfcore"
    tag "severus"
    tag "deepvariant"

    test("homo_sapiens - [ bam, bai, [], [], [] ], [[],[]]") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
                    [],
                    [],
                    []
                    ]
                input[1] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log.get(0).get(1)).readLines().last().contains("Writing VCF"),
                    process.out.read_qual,
                    process.out.breakpoints_double,
                    process.out.vcf,
                    process.out.breakpoints_clusters_list,
                    process.out.breakpoints_clusters,
                    process.out.plots
                    ).match() }
            )
        }

    }

    test("homo_sapiens - [ bam, bai, bam, bai, [] ], [[],[]]") {

        config "./nextflow.config"

        when {
            process {

                """
                input[0] = [
                    [ id:'test'], // meta map
                    file('/media/ssd_4tb/felix/projects/fellen31/modules/test2.sorted.bam', checkIfExists: true),
                    file('/media/ssd_4tb/felix/projects/fellen31/modules/test2.sorted.bam.bai', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true),
                    []
                ]
                input[1] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log.get(0).get(1)).readLines().last().contains("Writing VCF"),
                    process.out.read_qual,
                    process.out.breakpoints_double,
                    process.out.vcf,
                    process.out.breakpoints_clusters_list,
                    process.out.breakpoints_clusters,
                    process.out.plots
                    ).match() }
            )
        }

    }

    test("homo_sapiens - [ bam, bai, bam, bai, vcf ], [[],[]]") {

        config "./nextflow.config"

        setup {

            run("DEEPVARIANT") {
                script "../../deepvariant/main.nf"
                config "./deepvariant.config"

                process {
                    """
                    input[0] = [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.bed', checkIfExists: true)
                    ]
                    input[1] = [
                        [ id:'genome' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                    ]
                    input[2] = [
                        [ id:'genome' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)
                    ]
                    input[3] = [[],[]]
                    """
                }
            }

        }

        when {
            process {

                """
                input[0] = [
                    [ id:'test'], // meta map
                    file('/media/ssd_4tb/felix/projects/fellen31/modules/test2.sorted.bam', checkIfExists: true),
                    file('/media/ssd_4tb/felix/projects/fellen31/modules/test2.sorted.bam.bai', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true),
                    []
                ]
                input[1] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log.get(0).get(1)).readLines().last().contains("Writing VCF"),
                    process.out.read_qual,
                    process.out.breakpoints_double,
                    process.out.vcf,
                    process.out.breakpoints_clusters_list,
                    process.out.breakpoints_clusters,
                    process.out.plots
                    ).match() }
            )
        }

    }

    test("homo_sapiens - [ bam, bai, bam, bai, vcf ], [ bed ]Â ") {

        config "./nextflow.config"

        setup {

            run("DEEPVARIANT") {
                script "../../deepvariant/main.nf"
                config "./deepvariant.config"

                process {
                    """
                    input[0] = [
                        [ id:'test' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.bed', checkIfExists: true)
                    ]
                    input[1] = [
                        [ id:'genome' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                    ]
                    input[2] = [
                        [ id:'genome' ], // meta map
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)
                    ]
                    input[3] = [[],[]]
                    """
                }
            }

        }

        when {
            process {

                """
                input[0] = [
                    [ id:'test'], // meta map
                    file('/media/ssd_4tb/felix/projects/fellen31/modules/test2.sorted.bam', checkIfExists: true),
                    file('/media/ssd_4tb/felix/projects/fellen31/modules/test2.sorted.bam.bai', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true),
                    []
                ]
                input[1] = [
                    [ id:'bed'], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.bed', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    file(process.out.log.get(0).get(1)).readLines().last().contains("Writing VCF"),
                    process.out.read_qual,
                    process.out.breakpoints_double,
                    process.out.vcf,
                    process.out.breakpoints_clusters_list,
                    process.out.breakpoints_clusters,
                    process.out.plots
                    ).match() }
            )
        }

    }

    test("homo_sapiens - [ bam, bai, [], [], [] ], [[],[]] - stub") {

        config "./nextflow.config"
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/nanopore/bam/test.sorted.phased.bam.bai', checkIfExists: true),
                    [],
                    [],
                    []
                    ]
                input[1] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}