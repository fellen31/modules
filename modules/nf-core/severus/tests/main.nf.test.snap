{
    "homo_sapiens - [ bam, bai, [], [], [] ], [[],[]]": {
        "content": [
            false,
            [
                [
                    {
                        "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                    },
                    "read_qual.txt:md5,ec4218ff92346ef153d7a9c43e693dbb"
                ]
            ],
            [
                [
                    {
                        "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                    },
                    "breakpoints_double.csv:md5,f7e38adfba29aa770769207446fe9b37"
                ]
            ],
            null,
            null,
            null,
            null
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-14T18:56:45.492705081"
    },
    "homo_sapiens - [ bam, bai, bam, bai, vcf ], [ bed ]\u00a0": {
        "content": [
            false,
            [
                [
                    {
                        "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                    },
                    "read_qual.txt:md5,02f04c921926e209c0124c2210b301ab"
                ]
            ],
            [
                [
                    {
                        "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                    },
                    "breakpoints_double.csv:md5,1490d0ef4ced76dfff24abb01259a360"
                ]
            ],
            null,
            null,
            null,
            null
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-14T20:03:44.790184547"
    },
    "homo_sapiens - [ bam, bai, bam, bai, vcf ], [[],[]]": {
        "content": [
            false,
            [
                [
                    {
                        "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                    },
                    "read_qual.txt:md5,57847983bb1a28ad217bd01019e7bddb"
                ]
            ],
            [
                [
                    {
                        "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                    },
                    "breakpoints_double.csv:md5,1490d0ef4ced76dfff24abb01259a360"
                ]
            ],
            null,
            null,
            null,
            null
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-14T19:48:33.443319011"
    },
    "homo_sapiens - [ bam, bai, [], [], [] ], [[],[]] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "severus.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "read_qual.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "10": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        [
                            "severus_0.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "severus_1.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "11": [
                    
                ],
                "12": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_clusters_list.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "13": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_clusters.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "14": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        [
                            "severus_0.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "severus_1.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "15": [
                    "versions.yml:md5,0b91278f044f962709b734d204525398"
                ],
                "2": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_double.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "read_alignments:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "read_ids.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "severus_collaped_dup.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "severus_LOH.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "severus_all.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "8": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_clusters_list.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "9": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_clusters.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "all_breakpoints_clusters": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_clusters.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "all_breakpoints_clusters_list": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_clusters_list.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "all_plots": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        [
                            "severus_0.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "severus_1.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "all_vcf": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "severus_all.vcf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "breakpoints_double": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_double.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "collapsed_dup": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "severus_collaped_dup.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "severus.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "loh": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "severus_LOH.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "read_alignments": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "read_alignments:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "read_ids": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "read_ids.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "read_qual": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "read_qual.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "somatic_breakpoints_clusters": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_clusters.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "somatic_breakpoints_clusters_list": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        "breakpoints_clusters_list.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "somatic_plots": [
                    [
                        {
                            "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                        },
                        [
                            "severus_0.html:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "severus_1.html:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "somatic_vcf": [
                    
                ],
                "versions": [
                    "versions.yml:md5,0b91278f044f962709b734d204525398"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-14T20:13:33.203131531"
    },
    "homo_sapiens - [ bam, bai, bam, bai, [] ], [[],[]]": {
        "content": [
            false,
            [
                [
                    {
                        "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                    },
                    "read_qual.txt:md5,15f154776bbd9ae949bc4422070f5d9a"
                ]
            ],
            [
                [
                    {
                        "id": "test:md5,d41d8cd98f00b204e9800998ecf8427e"
                    },
                    "breakpoints_double.csv:md5,313dda4fef02d40509626a1d71440a01"
                ]
            ],
            null,
            null,
            null,
            null
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-14T19:40:41.274552715"
    }
}